examples/aggregator/scripts/deploy

```diff
@@ -29,6 +29,12 @@ then
   "${SCRIPTS_DIR}/undeploy" || true
 else
   "${SCRIPTS_DIR}/undeploy" -m || true
+
+  # Deleting the service does not immediately delete the node ports, so `deploy` fails 
+  # if it is run immediately after `undeploy`.
+  # Ref: https://github.com/kubernetes/kubernetes/issues/32987#issuecomment-344514151
+  echo "Waiting for 5 mins for k8s to remove all the old resources..."
+  sleep 5m
```

Comment:
BTW What's the story for more realistic deployments? Obviously we cannot undeploy every single time, since it's time consuming and also comes with downtime every time.

Reply:
Normally, we don't need the `-m` flag, which redeploys the metrics-sidecar. Without `-m`, the `deploy` command is very fast. I had to redeploy the metrics-sidecar because I had made some changes to the config (actually mistakenly! I had to make those changes to the aggregator container!). 

So, we'll hardly ever run `deploy` with the `-m` command. Though, it would be nice to run `sleep 5m` only if the `undeploy` command actually deletes some resources. Is it possible to return a value from a bash script?

I think more realistic deployments would retry the `deploy` if it fails.